contribs/gmf/src/controllers/abstract.js

```diff
@@ -147,11 +147,15 @@ gmf.AbstractController = function(config, $scope, $injector) {
     // Reload theme when login status changes.
     const previousThemeName = this.gmfThemeManager.getThemeName();
     this.gmfThemeManager.setThemeName('', true);
-    if (evt.type !== gmf.AuthenticationEventType.READY) {
-      this.updateCurrentTheme_(previousThemeName);
-    }
+
     // Reload themes and background layer when login status changes.
     this.gmfThemes_.loadThemes(roleId);
+
+    if (evt.type !== gmf.AuthenticationEventType.READY) {
+      this.gmfThemes_.getThemesObject().then(() => {
+        this.updateCurrentTheme_(previousThemeName);
+      });
+    }
```

Comment:
After a secound alalysis in can just be:
```suggestion
    if (evt.type !== gmf.AuthenticationEventType.READY) {
      this.updateCurrentTheme_(previousThemeName);
    }
```
Just change the order from the initial version ...

Reply:
According to my debugging this won't work. updateCurrentTheme_ happens still before the daa gets updated in the loadThemes promise.

Reply:
`updateCurrentTheme_` happens before the data gets of the theme, but in `updateCurrentTheme_` we have:
```js
this.gmfThemes_.getThemesObject().then((themes) => {
   // this should happens after we get the theme
```
And if this not working we should know why...

Reply:
I set my stops at a wrong point then. I deployed your simplified version on my problematic client's dev server and it works.